pkg/crc/machine/generate_bundle.go

```diff
@@ -0,0 +1,128 @@
+	"github.com/code-ready/crc/pkg/crc/oc"
+	crcssh "github.com/code-ready/crc/pkg/crc/ssh"
+	"github.com/code-ready/machine/libmachine/state"
+	"github.com/pkg/errors"
+)
+
+func (client *client) GenerateBundle() error {
+	bundleMetadata, sshRunner, err := loadVM(client)
+	if err != nil {
+		return err
+	}
+
+	err = removePullSecret(sshRunner)
+	if err != nil {
+		return errors.Wrap(err, "Error loading bundle metadata")
+	}
+
+	// Stop the cluster
+	currentState, err := client.Stop()
+	if err != nil {
+		return err
+	}
+	if currentState != state.Stopped {
+		return fmt.Errorf("VM is not stopped, current state is %s", currentState.String())
+	}
+
+	// Create a tmp directory with name of <bundle>_custom
+	tmpBaseDir, err := ioutil.TempDir("", "crc_custom_bundle")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpBaseDir)
+
+	// Create the custom bundle directory which is used as top level directory for tarball during compression
+	currentBundleName := bundle.GetBundleNameWithoutExtension(bundleMetadata.GetBundleName())
+	customBundleName := fmt.Sprintf("%s_custom", currentBundleName)
+	customBundleDir := filepath.Join(tmpBaseDir, customBundleName)
+
+	copier, err := bundle.NewCopier(bundleMetadata, tmpBaseDir, customBundleName)
+	if err != nil {
+		return err
+	}
+
+	if err := copier.CopyKubeConfig(); err != nil {
+		return err
+	}
+
+	kubePassword, err := cluster.GetKubeadminPassword(bundleMetadata)
+	if err != nil {
+		return err
+	}
+	if err := copier.SetKubeAdminPassword(kubePassword); err != nil {
+		return err
+	}
+
+	if err := copier.CopyPrivateSSHKey(constants.GetPrivateKeyPath()); err != nil {
+		return err
+	}
+
+	if err := copier.CopyOcExecutable(); err != nil {
+		return err
+	}
+
+	// Copy disk image
+	logging.Infof("Copy the disk image to %s", customBundleDir)
+	diskPath, err := copyDiskImage(customBundleDir, bundleMetadata.GetDiskImagePath())
+	if err != nil {
+		return err
+	}
+
+	if err := copier.SetDiskImage(diskPath); err != nil {
+		return err
+	}
+
+	return copier.GenerateBundle(customBundleName)
+}
+
+func loadVM(client *client) (*bundle.CrcBundleInfo, *crcssh.Runner, error) {
+	libMachineAPIClient, cleanup := createLibMachineClient()
+	defer cleanup()
+
+	host, err := libMachineAPIClient.Load(client.name)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Cannot load machine")
+	}
+
+	_, err = host.Driver.GetState()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Cannot get machine state")
+	}
+
+	crcBundleMetadata, err := getBundleMetadataFromDriver(host.Driver)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Error loading bundle metadata")
+	}
+
+	instanceIP, err := getIP(host, client.useVSock())
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Error getting the IP")
+	}
+	sshRunner, err := crcssh.CreateRunner(instanceIP, getSSHPort(client.useVSock()), crcBundleMetadata.GetSSHKeyPath(), constants.GetPrivateKeyPath(), constants.GetRsaPrivateKeyPath())
```

Comment:
yes it is same code we use it during the status and also in start, I think this key path should be part of ssh package itself where we create the runner and those should be default one and provide a way to add more to this default.  This might be a refactor and may be take care later.